File: backend/src/controllers/shareController.js
const Listing = require('../models/Listing');

// @desc    Paylaşım bilgisi oluştur
// @route   GET /api/share/:listingId
// @access  Public
const getShareInfo = async (req, res) => {
    try {
        const listing = await Listing.findById(req.params.listingId);
        if (!listing) {
            return res.status(404).json({ message: 'İlan bulunamadı' });
        }

        // Frontend URL'sine göre paylaşım linki oluştur
        const baseUrl = process.env.FRONTEND_URL || 'http://localhost:5173';
        const shareUrl = `${baseUrl}/listings/${listing._id}`;

        res.json({
            title: `${listing.brand} ${listing.model} - ${listing.year}`,
            description: `${listing.brand} ${listing.model} ${listing.year} | ${listing.km} km | ${listing.fuelType} | ${listing.price.toLocaleString('tr-TR')} TL`,
            url: shareUrl,
        });
    } catch (error) {
        res.status(500).json({ message: 'Sunucu hatası', error: error.message });
    }
};

module.exports = { getShareInfo };

File: backend/A2M2.API/Controllers/ShareController.cs
using A2M2.API.Services;
using Microsoft.AspNetCore.Mvc;

namespace A2M2.API.Controllers;

/// <summary>
/// Paylaşım controller'ı — Modül D: GET /cars/{carId}/share
/// </summary>
[ApiController]
[Route("api/cars")]
public class ShareController : ControllerBase
{
    private readonly ListingService _listingService;
    private readonly IConfiguration _config;

    public ShareController(ListingService listingService, IConfiguration config)
    {
        _listingService = listingService;
        _config = config;
    }

    /// <summary>GET /api/cars/{carId}/share — Paylaşım linki üret</summary>
    [HttpGet("{carId}/share")]
    public async Task<IActionResult> GetShareInfo(string carId)
    {
        var listing = await _listingService.GetByIdAsync(carId);
        if (listing == null) return NotFound(new { message = "İlan bulunamadı" });

        var frontendUrl = _config["AllowedOrigins:0"] ?? "http://localhost:5173";

        return Ok(new
        {
            title = $"{listing.Brand} {listing.Model} ({listing.Year})",
            description = $"{listing.Km:N0} km — {listing.Price:N0} ₺",
            url = $"{frontendUrl}/cars/{listing.Id}",
        });
    }
}
